### ISSUE TITLE
Bitbucket 8.9 not working with agent

### ISSUE THREAD

**Issue by voarsh2** [haxqer/jira#30]
TL:DR: I know you've only got repos for Jira/Confluence, but the agent *used to work for Bitbucket*. I attempted to install the DC Helm Chart for Bitbucket (as I use the agent successfully for Confluence/Jira), and 8.x+ (and 8.9 LTS throws an error when the agent is included).
7.21 LTS seems to not be working either

# Reproduce:

Seems bitbucket LTS 8.9 isn't working with the agent.

I placed the agent in the root on the home directory, used JVM args:
```
additionalJvmArgs:
    - '-javaagent:/var/atlassian/application-data/bitbucket/atlassian-agent.jar'
```

The error/output on 8.x when including the agent:

```
Starting Bitbucket webapp at http://localhost:7990
--
Wed, Jun 28 2023 6:44:46 pm | 17:44:46.272 [main] INFO com.atlassian.security.serialfilter.DeserializationFilterConfigurator - Global serial filter set to JDK 11 DeserializationFilter
Wed, Jun 28 2023 6:44:56 pm | 2023-06-28 17:44:56,791 INFO [main] c.a.b.i.b.BitbucketServerApplication Starting BitbucketServerApplication v8.9.2 using Java 11.0.19 on bitbucket-0 with PID 255 (/opt/atlassian/bitbucket/app/WEB-INF/classes started by bitbucket in /var/atlassian/application-data/bitbucket)
Wed, Jun 28 2023 6:44:56 pm | 2023-06-28 17:44:56,971 INFO [main] c.a.b.i.b.BitbucketServerApplication No active profile set, falling back to 1 default profile: "default"
Wed, Jun 28 2023 6:45:03 pm | 2023-06-28 17:45:02,549 INFO [main] c.a.b.i.boot.log.BuildInfoLogger Starting Bitbucket 8.9.2 (803e5fa built on Tue Jun 20 06:31:39 UTC 2023)
Wed, Jun 28 2023 6:45:03 pm | 2023-06-28 17:45:03,597 INFO [main] c.a.b.i.boot.log.BuildInfoLogger JVM: Eclipse Adoptium OpenJDK 64-Bit Server VM 11.0.19+7
Wed, Jun 28 2023 6:45:11 pm | 2023-06-28 17:45:11,666 INFO [main] c.a.b.i.b.BitbucketServerApplication Started BitbucketServerApplication in 24.58 seconds (JVM running for 31.314)
Wed, Jun 28 2023 6:45:31 pm | 2023-06-28 17:45:31,874 INFO [spring-startup] c.a.s.internal.home.HomeLockAcquirer Successfully acquired lock on home directory /var/atlassian/application-data/bitbucket
Wed, Jun 28 2023 6:45:52 pm | ============================== agent working ==============================
Wed, Jun 28 2023 6:45:52 pm | ============================== agent working ==============================
Wed, Jun 28 2023 6:45:52 pm | 2023-06-28 17:45:52,622 WARN [spring-startup] c.a.s.i.license.DefaultLicenseHelper The provided license could not be decoded
Wed, Jun 28 2023 6:45:52 pm | com.atlassian.extras.common.LicenseException: com.atlassian.extras.common.LicenseException: Failed to verify the license.
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.extras.decoder.v2.Version2LicenseDecoder.checkAndGetLicenseText(Version2LicenseDecoder.java:217)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.extras.decoder.v2.Version2LicenseDecoder.doDecode(Version2LicenseDecoder.java:114)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.extras.decoder.api.AbstractLicenseDecoder.decode(AbstractLicenseDecoder.java:15)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.extras.core.DefaultLicenseManager.decodeLicense(DefaultLicenseManager.java:62)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.extras.core.DefaultLicenseManager.getAtlassianLicense(DefaultLicenseManager.java:50)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.extras.core.DefaultLicenseManager.getLicense(DefaultLicenseManager.java:39)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.stash.internal.license.DefaultLicenseHelper.decode(DefaultLicenseHelper.java:67)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.stash.internal.license.DefaultLicenseHelper.getDecoded(DefaultLicenseHelper.java:108)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.stash.internal.license.DefaultLicenseHelper.isClusteringEnabled(DefaultLicenseHelper.java:117)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.stash.internal.home.SharedHomeLockAcquirer.isSharedLockRequired(SharedHomeLockAcquirer.java:106)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.stash.internal.home.SharedHomeLockAcquirer.createHomeLock(SharedHomeLockAcquirer.java:96)
Wed, Jun 28 2023 6:45:52 pm | at com.atlassian.stash.internal.home.HomeLockAcquirer.lock(HomeLockAcquirer.java:56)
Wed, Jun 28 2023 6:45:52 pm | at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Wed, Jun 28 2023 6:45:52 pm | at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Wed, Jun 28 2023 6:45:52 pm | at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Wed, Jun 28 2023 6:45:52 pm | at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:920)
Wed, Jun 28 2023 6:45:52 pm | at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
Wed, Jun 28 2023 6:45:52 pm | at javax.servlet.GenericServlet.init(GenericServlet.java:158)
Wed, Jun 28 2023 6:45:52 pm | at java.base/java.lang.Thread.run(Thread.java:829)
Wed, Jun 28 2023 6:45:52 pm | ... 37 frames trimmed
Wed, Jun 28 2023 6:45:52 pm | Caused by: com.atlassian.extras.common.LicenseException: Failed to verify the license.
```

7.21 LTS seems to not be working either

This is working fine for my confluence install (7.19 LTS)

Here's my example with confluence 7.19.10 LTS:

```
additionalContainers: []
additionalFiles: []
additionalHosts: []
additionalInitContainers: []
additionalLabels: {}
affinity: {}
confluence:
  accessLog:
    enabled: true
    localHomeSubPath: logs
    mountPath: /opt/atlassian/confluence/logs
  additionalBundledPlugins: []
  additionalEnvironmentVariables: []
  additionalJvmArgs:
    - '-javaagent:/var/atlassian/application-data/shared-home/libraries/atlassian-agent.jar'
    - '-Dsynchrony.service.url=https://xxx/synchrony/v1'

  license:
    secretKey: license-key
    secretName: conf-licence
 ```

**Comment by voarsh2** [comment]
After working on this a few hours and plenty of re-installs, I ran the .jar file inside the container.... and got a different licence🤷 .....

I am using 7.21.0 and there's no licence complaints. Will finish up this install, test it and see if upgrading to a later 7.21.x release works...........

**Comment by haxqer** [comment]
I have not personally used Bitbucket before. After conducting a test, I can confirm that the v8 version of Bitbucket cannot be cracked. However, the v7 version can be cracked and has already been compiled into an image, which can be found on [bitbucket](https://github.com/haxqer/bitbucket)

**Comment by haxqer** [comment]
Previously, I had not utilized Bitbucket. I had previously employed an internally deployed GitLab, but now I utilize GitHub.

**Comment by voarsh2** [comment]
> Previously, I had not utilized Bitbucket. I had previously employed an internally deployed GitLab, but now I utilize GitHub.
> 
> After conducting a test, I can confirm that the v8 version of Bitbucket cannot be cracked. However, the v7 version can be cracked and has already been compiled into an image, which can be found on [bitbucket](https://github.com/haxqer/bitbucket)
> 
> Thank you for providing the experimental data.！！！

That's disappointing, and surprising that V8 cannot be used with the crack.
Given that the original maintainer of the Atlassian Agent had their GitTea repo taken down, their last change was around 2021 (give or take) on the repo at that time, there's only a few "clone" repos of the original, so I am not sure if it's the absolute latest code of the original, so no one new has been working on the "crack". I heavily use Confluence and Jira, I thought it *might* be nice to replace my self-hosted GitLab instance with BitBucket for some tighter integration, and I needed to migrate my GitLab server to a completely new installation anyways, I thought I'd give it a try....

TL;DR: I hope someone can continue to work on the Atlassian Agent. I am not sure what changed in V8... and it makes me worry for Confluence/Jira, I am still on V7.X/8.X (Confluence/Jira respectively), I hope they don't break it..........

**Comment by voarsh2** [comment]
Not sure what's going on, but when I pass the agent in atlassian/bitbucket:7.21.0 I still get an error:

```
2023-06-30 13:24:54,158 INFO [spring-startup] c.a.s.internal.home.HomeLockAcquirer Successfully acquired lock on home directory /var/atlassian/application-data/bitbucket
--
Fri, Jun 30 2023 2:25:26 pm | ============================== agent working ==============================
Fri, Jun 30 2023 2:25:26 pm | ============================== agent working ==============================
Fri, Jun 30 2023 2:25:26 pm | 2023-06-30 13:25:26,921 INFO [spring-startup] c.a.s.internal.home.HomeLockAcquirer Successfully acquired lock on home directory /var/atlassian/application-data/bitbucket/shared
Fri, Jun 30 2023 2:25:29 pm | WARNING: An illegal reflective access operation has occurred
Fri, Jun 30 2023 2:25:29 pm | WARNING: Illegal reflective access by com.hazelcast.internal.networking.nio.SelectorOptimizer (file:/opt/atlassian/bitbucket/app/WEB-INF/lib/hazelcast-3.12.12.jar) to field sun.nio.ch.SelectorImpl.selectedKeys
Fri, Jun 30 2023 2:25:29 pm | WARNING: Please consider reporting this to the maintainers of com.hazelcast.internal.networking.nio.SelectorOptimizer
Fri, Jun 30 2023 2:25:29 pm | WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
Fri, Jun 30 2023 2:25:29 pm | WARNING: All illegal access operations will be denied in a future release
Fri, Jun 30 2023 2:25:29 pm | 2023-06-30 13:25:29,575 WARN [spring-startup] com.hazelcast.instance.Node [10.42.189.179]:5701 [bitbucket] [3.12.12] No join method is enabled! Starting standalone.
Fri, Jun 30 2023 2:26:16 pm | 2023-06-30 13:26:16,833 WARN [spring-startup] o.s.w.c.s.XmlWebApplicationContext Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'fileServerServlet' defined in URL [jar:file:/opt/atlassian/bitbucket/app/WEB-INF/lib/bitbucket-platform-7.21.0.jar!/com/atlassian/stash/internal/plugin/FileServerServlet.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'pluginResourceDownload' defined in class path resource [stash-context.xml]: Cannot resolve reference to bean 'contentTypeResolver' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'contentTypeResolver' defined in URL [jar:file:/opt/atlassian/bitbucket/app/WEB-INF/lib/bitbucket-platform-7.21.0.jar!/com/atlassian/stash/internal/plugin/SimpleContentTypeResolver.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.atlassian.stash.internal.plugin.SimpleContentTypeResolver]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: com/sun/activation/registries/LogSupport
Fri, Jun 30 2023 2:26:16 pm | 2023-06-30 13:26:16,952 INFO [spring-startup] c.a.s.internal.home.HomeLockAcquirer Releasing lock on /var/atlassian/application-data/bitbucket/shared
Fri, Jun 30 2023 2:26:16 pm | 2023-06-30 13:26:16,993 INFO [spring-startup] c.a.s.internal.home.HomeLockAcquirer Releasing lock on /var/atlassian/application-data/bitbucket
Fri, Jun 30 2023 2:26:17 pm | 2023-06-30 13:26:17,002 ERROR [spring-startup] c.a.j.s.w.s.JohnsonDispatcherServlet SpringMVC dispatcher [springMvc] could not be started
Fri, Jun 30 2023 2:26:17 pm | org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'fileServerServlet' defined in URL [jar:file:/opt/atlassian/bitbucket/app/WEB-INF/lib/bitbucket-platform-7.21.0.jar!/com/atlassian/stash/internal/plugin/FileServerServlet.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'pluginResourceDownload' defined in class path resource [stash-context.xml]: Cannot resolve reference to bean 'contentTypeResolver' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'contentTypeResolver' defined in URL [jar:file:/opt/atlassian/bitbucket/app/WEB-INF/lib/bitbucket-platform-7.21.0.jar!/com/atlassian/stash/internal/plugin/SimpleContentTypeResolver.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.atlassian.stash.internal.plugin.SimpleContentTypeResolver]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: com/sun/activation/registries/LogSupport
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
Fri, Jun 30 2023 2:26:17 pm | at javax.servlet.GenericServlet.init(GenericServlet.java:158)
Fri, Jun 30 2023 2:26:17 pm | at java.base/java.lang.Thread.run(Thread.java:829)
Fri, Jun 30 2023 2:26:17 pm | ... 18 frames trimmed
Fri, Jun 30 2023 2:26:17 pm | Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'pluginResourceDownload' defined in class path resource [stash-context.xml]: Cannot resolve reference to bean 'contentTypeResolver' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'contentTypeResolver' defined in URL [jar:file:/opt/atlassian/bitbucket/app/WEB-INF/lib/bitbucket-platform-7.21.0.jar!/com/atlassian/stash/internal/plugin/SimpleContentTypeResolver.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.atlassian.stash.internal.plugin.SimpleContentTypeResolver]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: com/sun/activation/registries/LogSupport
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:342)
Fri, Jun 30 2023 2:26:17 pm | ... 5 common frames omitted
Fri, Jun 30 2023 2:26:17 pm | Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'contentTypeResolver' defined in URL [jar:file:/opt/atlassian/bitbucket/app/WEB-INF/lib/bitbucket-platform-7.21.0.jar!/com/atlassian/stash/internal/plugin/SimpleContentTypeResolver.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.atlassian.stash.internal.plugin.SimpleContentTypeResolver]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: com/sun/activation/registries/LogSupport
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1334)
Fri, Jun 30 2023 2:26:17 pm | ... 5 common frames omitted
Fri, Jun 30 2023 2:26:17 pm | Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.atlassian.stash.internal.plugin.SimpleContentTypeResolver]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: com/sun/activation/registries/LogSupport
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:224)
Fri, Jun 30 2023 2:26:17 pm | ... 5 common frames omitted
Fri, Jun 30 2023 2:26:17 pm | Caused by: java.lang.NoClassDefFoundError: com/sun/activation/registries/LogSupport
Fri, Jun 30 2023 2:26:17 pm | at javax.activation.MimetypesFileTypeMap.<init>(MimetypesFileTypeMap.java:130)
Fri, Jun 30 2023 2:26:17 pm | at javax.activation.MimetypesFileTypeMap.<init>(MimetypesFileTypeMap.java:301)
Fri, Jun 30 2023 2:26:17 pm | at com.atlassian.stash.internal.plugin.SimpleContentTypeResolver.<init>(SimpleContentTypeResolver.java:25)
Fri, Jun 30 2023 2:26:17 pm | at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
Fri, Jun 30 2023 2:26:17 pm | at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
Fri, Jun 30 2023 2:26:17 pm | at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
Fri, Jun 30 2023 2:26:17 pm | at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:211)
Fri, Jun 30 2023 2:26:17 pm | ... 5 common frames omitted
Fri, Jun 30 2023 2:26:17 pm | Caused by: java.lang.ClassNotFoundException: com.sun.activation.registries.LogSupport
Fri, Jun 30 2023 2:26:17 pm | at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
Fri, Jun 30 2023 2:26:17 pm | at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
Fri, Jun 30 2023 2:26:17 pm | ... 12 common frames omitted
Fri, Jun 30 2023 2:26:18 pm | 2023-06-30 13:26:17,908 WARN [http-nio-7990-exec-10] c.a.j.s.w.f.BypassableDelegatingFilterProxy Bypassing [hazelcastFilter]; no Spring WebApplicationContext is available
Fri, Jun 30 2023 2:26:18 pm | 2023-06-30 13:26:18,141 WARN [http-nio-7990-exec-10] c.a.j.s.w.f.BypassableDelegatingFilterProxy Bypassing [requestAttributeFilter]; no Spring WebApplicationContext is available
Fri, Jun 30 2023 2:26:18 pm | 2023-06-30 13:26:18,142 WARN [http-nio-7990-exec-10] c.a.j.s.w.f.BypassableDelegatingFilterProxy Bypassing [entityMovedFilter]; no Spring WebApplicationContext is available
Fri, Jun 30 2023 2:26:18 pm | 2023-06-30 13:26:18,142 WARN [http-nio-7990-exec-10] c.a.j.s.w.f.BypassableDelegatingFilterProxy Bypassing [pluginProvidedAuthenticationFilter]; no Spring WebApplicationContext is available
Fri, Jun 30 2023 2:26:18 pm | 2023-06-30 13:26:18,142 WARN [http-nio-7990-exec-10] c.a.j.s.w.f.BypassableDelegatingFilterProxy Bypassing [trustedApplicationsAuthenticationFilter]; no Spring WebApplicationContext is available
Fri, Jun 30 2023 2:26:18 pm | 2023-06-30 13:26:18,143 WARN [http-nio-7990-exec-10] c.a.j.s.w.f.BypassableDelegatingFilterProxy Bypassing [goGetFilter]; no Spring

```

But will now try building the docker image myself from your repo.

**Comment by voarsh2** [comment]
There is a new repo that I found that fixes the agent for Bitbucket 8.9, I have confirmed by building the agent with maven and running the 8.9 LTS Helm Chart! :)

Let me know if you need help with building, or I can give you the build I made:
https://github.com/caelumlux/atlassian-agent

**Comment by haxqer** [comment]
Wow, that's very impressive. Thank you for providing the information

**Comment by voarsh2** [comment]
Sorry for double posting. Will close this issue now. Again let me know if you need help building it.
I'm going to assume that future builds will follow Bitbucket and will break your 1.3.3 Agent soon.
